### PR TITLE
ROC-2984: Add configuration class to toggle on or off

### PR DESCRIPTION
**Pull request by sabahirfan**
This PR adds the `Configuration` class to conditionally enable health check indicator.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '1.0.9'
+version '1.0.10'
```
Comment (geek-soft) [inline build.gradle]: I would bump second number as this change is a bit breaking.

File: src/main/resources/META-INF/spring.factories [hunk 2]
```diff
@@ -1,2 +1,2 @@
 org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
-  uk.gov.hmcts.reform.ccd.client.healthcheck.CoreCaseDataHealthIndicator
+  uk.gov.hmcts.reform.ccd.client.CoreCaseDataClientConfiguration
```
Comment (geek-soft) [inline src/main/resources/META-INF/spring.factories]: I would name class `CoreCaseDataClientAutoConfiguration` as this seems to be spring convention.

See fo reference:

https://github.com/spring-projects/spring-boot/blob/v1.5.9.RELEASE/spring-boot-autoconfigure/src/main/resources/META-INF/spring.factories

File: src/main/java/uk/gov/hmcts/reform/ccd/client/CoreCaseDataClientConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.context.annotation.Configuration;
+import uk.gov.hmcts.reform.ccd.client.healthcheck.CoreCaseDataHealthIndicator;
+
+@Configuration
+@EnableFeignClients(basePackages = "uk.gov.hmcts.reform.ccd.client")
+public class CoreCaseDataClientConfiguration {
+
+    @Bean
+    @ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
+    public CoreCaseDataHealthIndicator coreCaseDataHealthIndicator(CoreCaseDataApi coreCaseDataApi) {
```
Comment (geek-soft) [inline src/main/java/uk/gov/hmcts/reform/ccd/client/CoreCaseDataClientConfiguration.java]: Can we name it just `healthIndicator`? I guess that enough to say in the context of the CCD client library.

File: build.gradle [hunk 4]
```diff
@@ -10,7 +10,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '1.0.9'
+version '1.1.0'
```
Comment (rlewan) [inline build.gradle]: 👍  🙂

**Pull request merged**